server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -856,11 +858,46 @@ int getAllTimeseriesCount(PartialPath prefixPath) throws MetadataException {
     if (nodes.length == 0 || !nodes[0].equals(root.getName())) {
       throw new IllegalPathException(prefixPath.getFullPath());
     }
+    Holder<Integer> countHolder = new Holder<>(0);
     try {
-      return getCount(root, nodes, 1);
+      getCount(root, nodes, 1, countHolder, false);
     } catch (PathNotExistException e) {
       throw new PathNotExistException(prefixPath.getFullPath());
     }
+    return countHolder.getValue();
+  }
+
+  /** Traverse the MTree to get the count of timeseries. */
+  private void getCount(
+      MNode node, String[] nodes, int idx, Holder<Integer> countHolder, boolean wildcard)
```

Comment:
Holder can be avoided just define the function as `private int getCount`

Reply:
ok, i remove holder